Meadowlark-js/tests/e2e/ResourcesReferenceValidation.test.ts

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+import { getAccessToken, Clients } from './functions/Credentials';
+import { createResource, deleteResourceByLocation } from './functions/Resources';
+import { createSchool } from './functions/Shared';
+import { baseURLRequest, rootURLRequest } from './Setup';
+
+describe('When creating a resource that has a reference to another resource', () => {
+  describe('given a token with strict validation', () => {
+    describe('given reference does not exist', () => {
+      it('should fail with code 400 and a message', async () => {
+        await baseURLRequest
+          .post('/v3.3b/ed-fi/locations')
+          .auth(await getAccessToken(Clients.Vendor1), { type: 'bearer' })
+          .send({
+            classroomIdentificationCode: 'string',
+            schoolReference: {
+              schoolId: 99,
+            },
+            maximumNumberOfSeats: 20,
+            optimalNumberOfSeats: 10,
+          })
+          .expect(400)
+          .then((response) => {
+            expect(response.body.message).toContain('Resource School is missing identity');
```

Comment:
That's a terrible error message and we'll need to change it. Let's leave it alone for right now, but just be aware that it will change in the future. For the next round of tests, might want to simply use `expect(response.body).toHaveProperty('message')` (I think that's the syntax). No need to change right now though.